torch/_library/utils.py

```diff
@@ -156,3 +157,67 @@ def zip_schema(schema, args, kwargs):
             continue
         yield info, args[i]
     return
+
+
+def mangle(value: str) -> str:
+    """Mangles a string into a usable Python identifier.
+
+    https://docs.python.org/3/reference/lexical_analysis.html#identifiers
+
+    The string may consist of letters, numbers, underscores and [.<>@]
+
+    We optimize for readability in the general case (strings with letters,
+    numbers, periods, and underscore).
+
+    The mangling scheme is:
+    - . -> X
+    - Z -> ZZ
+    - X -> ZX
+    - < -> Z0
+    - > -> Z1
+    - @ -> Z2
+
+    Examples:
+    - foo.bar.baz -> fooXbarXbaz
+    - foo_bar.baz -> foo_barXbaz
+    - X.Z -> ZXXZZ
+    - ZQ.ZQ -> ZZQXZZQ
+    - <lambda> -> Z0lambdaZ1
+    """
+    assert re.match(r"^[A-z0-9.<>@_]+$", value), "invalid value"
+    value = value.replace("Z", "ZZ")
+    value = value.replace("X", "ZX")
+    value = value.replace(".", "X")
+    value = value.replace("<", "Z0")
+    value = value.replace(">", "Z1")
+    value = value.replace("@", "Z2")
+    return value
+
+
+def demangle(value: str) -> str:
+    """The inverse operation of mangle"""
+    value = value.replace("ZZ", "!")
+    value = value.replace("ZX", "?")
+    value = value.replace("Z0", "<")
+    value = value.replace("Z1", ">")
+    value = value.replace("Z2", "@")
+    value = value.replace("X", ".")
+    value = value.replace("?", "X")
+    value = value.replace("!", "Z")
+    return value
+
+
+def unique_name(fn: Callable) -> str:
+    """Given a callable, return a name that uniquely identifies it.
+
+    Some cases:
+    - a function is identified by its __module__ and __qualname__
+    - lambda requires the object's id to distinguish
+    """
+    is_lambda = fn.__name__ == "<lambda>"
```

Comment:
nit: check that the input is actually a callable?
Also is a Callable guaranteed to have a `__name__`/`__module__`/`__qualname__` attributes?